omniledger/service/service.go

```diff
@@ -622,6 +638,37 @@ clientTransactions:
 	return cdbTemp.GetRoot(), ctsOK, states, nil
 }
 
+func (s *Service) getLeader(scID skipchain.SkipBlockID) (*network.ServerIdentity, error) {
+	sb, err := s.db().GetLatestByID(scID)
+	if err != nil {
+		return nil, err
+	}
+	if sb.Roster == nil || len(sb.Roster.List) < 1 {
+		return nil, errors.New("roster is empty")
+	}
+	return sb.Roster.List[0], nil
+}
+
+func (s *Service) getTxs(leader *network.ServerIdentity, scID skipchain.SkipBlockID) ClientTransactions {
```

Comment:
Why does getTxs even care who the current leader is?

Reply:
So it can verify if it is the _real_ leader?